main/src/cgeo/geocaching/settings/Settings.java

```diff
@@ -821,6 +821,10 @@ public static int getMapLineValue(final int prefKeyId, final int defaultValueKey
         return getInt(prefKeyId, getKeyInt(defaultValueKeyId));
     }
 
+    public static int getMapOsmThreads() {
+        return getBoolean(R.string.pref_map_osm_multithreaded, false) ? Math.max(1, getInt(R.string.pref_map_osm_threads, Math.min(Runtime.getRuntime().availableProcessors() + 1, 4))) : 1;
```

Comment:
Why are you using the hard-coded number 4 as limit for multithreading, and not always the available number of processors?

Reply:
To limit resource usage, weighting additional memory use vs. incremental rendering performance gain.